kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CommitterImpl.java

```diff
@@ -0,0 +1,253 @@
+    if (isLeader && coordinatorThread.isTerminated()) {
+      throw new IllegalStateException("Coordinator unexpectedly terminated");
+    }
+  }
+
+  @Override
+  public void commit(CommittableSupplier committableSupplier) {
+    throwExceptionIfCoordinatorIsTerminated();
+
+    Optional<UUID> maybeCommitId = commitRequestListener.getCommitId();
+    if (maybeCommitId.isPresent()) {
+      UUID commitId = maybeCommitId.get();
+
+      Committable committable = committableSupplier.committable();
+
+      List<Event> events = Lists.newArrayList();
+
+      committable
+          .writerResults()
+          .forEach(
+              writerResult -> {
+                Event commitResponse =
+                    new Event(
+                        config.controlGroupId(),
+                        EventType.COMMIT_RESPONSE,
+                        new CommitResponsePayload(
+                            writerResult.partitionStruct(),
+                            commitId,
+                            TableName.of(writerResult.tableIdentifier()),
+                            writerResult.dataFiles(),
+                            writerResult.deleteFiles()));
+
+                events.add(commitResponse);
+              });
+
+      // include all assigned topic partitions even if no messages were read
+      // from a partition, as the coordinator will use that to determine
+      // when all data for a commit has been received
+      List<TopicPartitionOffset> assignments =
+          context.assignment().stream()
+              .map(
+                  topicPartition -> {
+                    Offset offset =
+                        committable.offsetsByTopicPartition().getOrDefault(topicPartition, null);
+                    return new TopicPartitionOffset(
+                        topicPartition.topic(),
+                        topicPartition.partition(),
+                        offset == null ? null : offset.offset(),
+                        offset == null ? null : offset.timestamp());
+                  })
+              .collect(toList());
+
+      Event commitReady =
+          new Event(
+              config.controlGroupId(),
+              EventType.COMMIT_READY,
+              new CommitReadyPayload(commitId, assignments));
+      events.add(commitReady);
+
+      Map<TopicPartition, OffsetAndMetadata> offsetsToCommit =
+          committable.offsetsByTopicPartition().entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey, e -> new OffsetAndMetadata(e.getValue().offset())));
+
+      KafkaUtils.sendAndCommitOffsets(
+          producerId,
+          producer,
+          config.controlTopic(),
+          events,
+          offsetsToCommit,
+          new ConsumerGroupMetadata(config.controlGroupId()));
+
+      KafkaUtils.commitOffsets(
+          producer, offsetsToCommit, new ConsumerGroupMetadata(config.connectGroupId()));
```

Comment:
This is where I commit to the `connect-` group now. 
Right next to where I commit to the (current) source-of-truth group: control-group-id. 
Note that committing to the `connect-` group is best-effort (same as before); the connect group is not currently the source of truth. 
In a future PR, we will start committing to the `connect-` group exclusively and make that the source of truth, however that is a breaking change which I want to avoid in this PR (we should bundle up breaking changes in another PR).

Reply:
Re; future PR, if we stop committing to `connect-` won't we run into the potential for the `connect-` consumer group to be deleted by Kafka due to inactivity?

Reply:
I think you've misunderstood. In the future, we will still be committing to the `connect-` consumer group. The only change is that we will be committing to the `connect-` group **_exclusively_** i.e. I intend to get rid of the connector-managed-consumer-group (but this is all theoretical right now, we can talk about it more when we get there).